### ISSUE TITLE
link_documents plugin conflict with check_closing plugin

### ISSUE THREAD

**Issue by upsuper** [beancount/fava#1301]
This is a minimal reproducible beancount:
```beancount
option "operating_currency" "AUD"

plugin "beancount.plugins.check_closing"
plugin "fava.plugins.link_documents"

2021-06-30 open Equity:Opening-Balances
2021-06-30 open Assets:X    AUD
2021-06-30 open Expenses:Y  AUD
2021-06-30 pad Assets:X Equity:Opening-Balances
2021-07-01 balance Assets:X                   100.00 AUD

2021-07-01 document Assets:X "a.pdf"
2021-07-01 * "Something"
    document: "a.pdf"
    Assets:X                                 -100.00 AUD
        closing: TRUE
    Expenses:Y
```

Loading it in Fava, you can see that the document is not linked to the transaction as expected:
![image](https://user-images.githubusercontent.com/333750/123801631-d11c9880-d92d-11eb-90ea-0676b98bc1a6.png)
also if you open the transaction details you would see it's not showing the right range of lines:
![image](https://user-images.githubusercontent.com/333750/123801761-ebef0d00-d92d-11eb-83c4-5400bcc72012.png)

**Comment by yagebu** [comment]
The `check_closing` replaces the transaction metadata with the posting metadata (minus the `closing: True`) at https://github.com/beancount/beancount/blob/f13f6d058f1ea5e2e6677e649cebcc9204469d37/beancount/plugins/check_closing.py#L63, so in your case it removes the `document` part of the metadata dictionary. Nothing the link_documents plugin can really do after that. Maybe that's a bug in the check_closing plugin. A workaround might be to change the order of the plugins (although then you'd still miss the document metadata in the journal view).

**Comment by upsuper** [comment]
Indeed changing the order doesn't fix the issue. I wonder why it has to replace the meta...

**Comment by upsuper** [comment]
Alright, submitted a PR to fix beancount plugin: beancount/beancount#664. We can probably close this, as it's not a Fava issue...
